src/other/utils.jl

```diff
@@ -477,3 +477,22 @@ function _findall(B::BitVector)::Union{UnitRange{Int}, Vector{Int}}
     end
     @assert false "should not be reached"
 end
+
+# Creates a vector like `vect` without the missing type and copies only data at indices in `inds`
+# Fuses two operations that were previously done by getindex + disallowmissing
+# unsafe because it does not check for `inds` values to be not missing (ensured by the caller)
+# Called by `_getindex_disallowmissing(df,...)`
+function _getindex_disallowmissing(vect::AbstractVector{T}, inds::AbstractVector{<:Integer}) where {T}
+    # General fallback, relies on custom vectors' own methods
+    return vect[inds]|>disallowmissing
```

Comment:
Let us continue the discussion with this inner function as it is a core.
I have benchmarked your proposal against doing `disallowmissing(@view vect[inds])` and it seems that using `@view` is comparable to your faster option (while being also relatively fast for your fallback option).

Can you please check it?

(after we finalize the "inner method", we will go back to general design.

Thank you!

Reply:
Of course, you're right -- that is much more elegant and much faster! I wish I didn't waste as much time on the for-loops :)

These are the benchmarking results - it scales pretty well and this is for numeric columns! 
(the above proposal is named `_getindex_disallowmissing_bk`)
<img width="996" alt="image" src="https://user-images.githubusercontent.com/49557684/213864199-9259cbb3-1329-48b9-9805-64f6ee3b7d40.png">

It performs better than the default for "pooled" arrays and it's within c. 10-15% of the numeric columns, which is great.

I went ahead and changed the PR to reflect this approach.

***
Would the next be to write a testset for the new inner function?

Reply:
So now I think that what should be done is change the PR so that the new function is invoked and run by standard tests.

Two general comments:
1. The `disallowmissing(@view vect[inds])` is so short that I do not think we need a separate function for it.
2. It seems that we do not have to have a separate code path for `AbstractDataFrame`. All could go into one function but just in place where you use `_columns` for `DataFrame`, `collect(eachcol(...` can be used for other data frame types
3. In several places you use `@inbounds` but I do not think they are needed (i.e. they are correct, but they should not affect the performance - in such cases it is better to use safer constructs and not use `@inbounds`)

In summary - I would propose to clean up the PR so that the code has a final form (and then I will review it) and make sure it passes existing tests (of course it is OK to add new tests now also).

Reply:
Noted! I have done the cleanup as you suggested and the tests PASSED.

Re 1) Agreed. The only potential benefit could have been to ensure handling of all column types and ensuring "proper" return type 

Re 2) Agreed. I've made the change, but it comes at the cost of varying levels of abstraction (all other conditional branches are fairly "light"). 
I've tried to simplify the nested IFs, but I'm not sure how to go any further to ensure proper behaviour (?threaded, ?disallowmissing

Re 3) Removed it. I'm not sure why it doesn't matter, because there is still indexing and hence bounds checks should be happening, but since there seems to be no penalty, I've removed it.

Reply:
One random note, please have a look at pathway for dropmissing(df::SubDataFrame)

The implemented behaviour is that dropmissing(::AbstractDF) will always produce a DataFrames (if view=false).

I'm not sure how `disallowmissing!` was implemented for SubDataFrame (couldn't find the method), but practically from trying it out `dropmissing(::SubDataFrame) returned DataFrame. So practically there shouldn't be any change (if I got it right).